pkg/client/unversioned/clientcmd/loader.go

```diff
@@ -332,7 +332,17 @@ func LoadFromFile(filename string) (*clientcmdapi.Config, error) {
 	// set LocationOfOrigin on every Cluster, User, and Context
 	for key, obj := range config.AuthInfos {
 		obj.LocationOfOrigin = filename
+		if len(obj.Token) == 0 && len(obj.TokenFile) > 0 {
```

Comment:
Ok, I can do this, though the consistent change is likely to be construed as a breaking change in the API (so we'll need to support both for a while...)

I'll send the updated code later today.